src/modeling/LLM_queries.py

```diff
@@ -0,0 +1,108 @@
+import ollama 
+from textwrap import dedent
+import math
+import re
+
+
+# models = ['llama3.1:latest', 'aminadaven/dictalm2.0-instruct:f16']
+def summarize(col_name: str, sentences: str, model='aminadaven/dictalm2.0-instruct:f16', verbose=0): # model in 'llama3'
+    
+    # system_instructions = ' '.join([
+    #     f"המטרה שלך היא לתרגם לעברית ואז לסכם במדויק בעברית.",
+    #     f"מצורפות תשובות התלמידים לגבי השאלה הבאה [<'{col_name.strip()}'>].",
+    # f"סכם את התשובות בפסקה אחת בדיוק עם לכל היותר {max(math.ceil(math.log(len(sentences))),2)} משפטים.",
+    # "הקפד לוודא שהתשובה מבוססת רק על הדעות שניתנו.",
+    # "מבחינה דקדוקית, נסח את הסיכום בגוף ראשון יחיד, כאילו אתה אחד הסטודנטים.",
+    # "כתוב את הסיכום בעברית בלבד, ללא תוספות לפני או אחרי הסיכום",
+    # "בתשובתך אל תוסיף על הקלט ואל תגיב ואל תביע את דעתך. רק סכם בהתבסס על הקלט מהסטודנטים בלבד."
+    # ])
+    # system_instructions = dedent(f"""\
+    #                 You are a student in an arabic language course. 
+    #                 Your task is return concise summary of the opinion of the other students.
+    #                 The students are answering the question: [<'{col_name.strip()}'>]
+    #                 You are given list of students' opinions, separated by |, Summarize their opinions. 
+    #                 Format your response with up to {min(max(math.ceil(math.log(len(sentences))),2), 5)} short sentences, phrased like a student's opinion.
+    #                 Your response must be in English.
+    #                 Do not answer the students, do not add your opinion, do not add comments before or after the summary. Just summarize!
+    #                 """)
+    num_output_sentences_ = min(max(math.ceil(math.log(len(sentences))),2), 5)
+    system_instructions = dedent(f"""\
```

Comment:
we maybe better extract prompts to external file, so we can modify it there while working on code in parallel.